appinventor/misc/checkstyle/checkstyle-git.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+import subprocess
+from io import StringIO
+from re import compile
+import os
+import sys
+
+
+parent_commit = 'HEAD' if len(sys.argv) < 2 else sys.argv[1]
+checkstyle_pattern = compile(r'^\[(?P<severity>[^\]]*)\] (?P<filename>[^:]*):(?P<line>[0-9]*):(?P<message>.*)'
+                             r' \[(?P<rule>[^\]]*)\]$')
+compile(r'^\[(?P<severity>[^\]]*)\]')
+basedir = os.getcwd() if os.path.exists('build.xml') else os.path.join(os.getcwd(), 'appinventor')
+
+
+def checkstyle(current_file):
+    return StringIO(subprocess.check_output(['java', '-jar', 'lib/checkstyle/checkstyle.jar', '-c',
+                                             'lib/checkstyle/appinventor_checks.xml', current_file],
+                                            encoding='utf-8', cwd=basedir))
+
+
+def check_chunks(checkstyle_output, chunks):
+    """
+
+    :param checkstyle_output:
+    :param chunks:
+    :type chunks: list[(int, int)]
+    :return:
+    """
+    success = True
+    i = 0
+    for line in checkstyle_output:
+        match = checkstyle_pattern.match(line.strip())
+        if match:
```

Comment:
This line can use the new Walrus operator `:=` introduced with Python 3.8

https://www.python.org/dev/peps/pep-0572/

---
You can choose to ignore this one but I couldn't resist 😄